packages/node/src/lib/log.ts

```diff
@@ -0,0 +1,145 @@
+import type { LogOptions } from './construct-payload';
+import type { GroupingObject, OutgoingLogBody } from './metrics-log';
+import type { Request, Response } from 'express';
+import config from '../config';
+import clamp from 'lodash/clamp';
+import * as url from 'url';
+import { v4 as uuidv4 } from 'uuid';
+import { constructPayload } from './construct-payload';
+import { metricsAPICall } from './metrics-log';
+
+let queue: OutgoingLogBody[] = [];
+function doSend(readmeApiKey, options) {
+  // Copy the queue so we can send all the requests in one batch
+  const json = queue.slice();
+  // Clear out the queue so we don't resend any data in the future
+  queue = [];
+
+  // Make the log call
+  metricsAPICall(readmeApiKey, json).catch(e => {
+    // Silently discard errors and timeouts.
+    if (options.development) throw e;
+  });
+}
+// Make sure we flush the queue if the process is exited
+process.on('exit', doSend);
+
+interface ExtendedResponse extends Response {
+  _body?: string;
+}
+
+// We're doing this to buffer up the response body
+// so we can send it off to the metrics server
+// It's unfortunate that this isn't accessible
+// natively. This may take up lots of memory on
+// big responses, we can make it configurable in future
+function patchResponse(res) {
```

Comment:
wish we could find something better for this :/

Reply:
Yeah I've tried, I really have. `res` is a [ServerResponse](https://nodejs.org/docs/latest/api/http.html#class-httpserverresponse) which is a WriteableStream. It can have stuff piped to it, but there's no way to pipe that somewhere else or listen to it. Would love for you to see if there's a better way around this!